Upgrade to Babel 6.x

fabriceci opened the pull request:
current Babel.rc's file was not compatible with Babel 6+:

```
ERROR in ./node_modules/attr-accept/dist/index.js
Module build failed: ReferenceError: [BABEL] /node_modules/attr-accept/dist/index.js: Using removed Babel 5 option - Check out the corresponding stage-x presets http://babeljs.io/docs/plugins/#presets
`````

okonet commented:
This package still uses Babel 5: https://github.com/okonet/attr-accept/blob/master/package.json#L34

so I'm not sure why you're getting this error.

fabriceci commented:
Oh silly me, update the babel file is not enough indeed.

Babel 5 is deprecated, if you build a project with a dependency that use Babel 5 (it's my case), you will see this error.

I updated to babel 6 as well as all the dependencies (except for "semantic-release" because I don't know this package).

I also added Prettier to format the code.

okonet commented:
@fabriceci thanks for working on this. Let's separate prettier and babel updates since prettier will require updating `.eslintrc` and `.editorconfig` to remove conflicting rules. I'll do this part myself.

okonet commented on .eslintrc:
```diff
@@ -1,10 +1,11 @@
 {
-  "extends": "airbnb",
+  "extends": ["airbnb", "prettier"],
```
Let's skip prettier in this PR please

fabriceci commented:
The 2 changes in "index.js" are due to Eslint default rules.

okonet commented:
Since CI is failing for Node < 4.x I'd propose to set the `"engines": ">= 4"` in `package.json` and update `.travis.yml` to not run on previous versions.

fabriceci commented:
I updated the file, the travis checks have passed.

okonet commented:
Cool! Thanks for working on it.

okonet commented:
Sorry I had to revert it since:

1. I messed up when merging and merged single commits that doesn't follow commitizen changeling format
1. You aded `yarn.lock` file.

Could you please rebase on top latest master, remove the `yarn.lock` and resubmit this PR?

okonet commented:
Please note that I've force pushed so the master is on the commit you've branched from.
